sklearn/datasets/tests/test_svmlight_format.py

```diff
@@ -553,3 +553,27 @@ def test_load_offset_exhaustive_splits():
 def test_load_with_offsets_error():
     with pytest.raises(ValueError, match="n_features is required"):
         _load_svmlight_local_test_file(datafile, offset=3, length=3)
+
+
+def test_multilabel_y_explicit_zeros(tmp_path):
+    """
+    Ensure that if y contains explicit zeros (i.e. elements of y.data equal to
+    0) then those explicit zeros are not encoded.
+    """
+    save_path = str(tmp_path / "svm_explicit_zero")
+    rng = np.random.RandomState(42)
+    X = rng.randn(3, 5).astype(np.float64)
+    indptr = np.array([0, 2, 3, 6])
+    indices = np.array([0, 2, 2, 0, 1, 2])
+    data = np.array([0, 2, 3, 4, 5, 6])  # The first element is an explicit zero
```

Comment:
The [Multilabel format](https://scikit-learn.org/stable/modules/multiclass.html#multiclass-and-multioutput-algorithms) that scikit-learn expects are ones and zeros.

Also to make this more interesting, lets set the last element to zero:

```suggestion
    # The first and last element is an explicit zero
    data = np.array([0, 1, 1, 1, 1, 0])
```